src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -351,6 +356,16 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
     this._assertFormFieldControl();
   }
 
+  ngAfterViewChecked() {
+    // After the appearance changes to `outline`, measure the size of the label and provide it
+    // to the notched outline.
+    if (this._needsOutlineNotchWidthRefresh) {
+      this._refreshOutlineNotchWidth();
+      this._changeDetectorRef.detectChanges();
+      this._needsOutlineNotchWidthRefresh = false;
```

Comment:
Couldn't we do this either inside the `appearance` setter or in `ngOnChanges`?

Reply:
`ngOnChanges` probably would have been a better place for this - but removed due to Paul's suggestion